src/inmanta/data/sqlalchemy.py

```diff
@@ -0,0 +1,640 @@
+"""
+Copyright 2025 Inmanta
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+Contact: code@inmanta.com
+"""
+
+import uuid
+from typing import List, Optional
+
+from inmanta.data.model import EnvSettingType
+from sqlalchemy import Boolean, ForeignKeyConstraint, Index, Integer, PrimaryKeyConstraint, String, UniqueConstraint, text
+from sqlalchemy.dialects.postgresql import JSONB, UUID
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column, relationship
+
+
+class Base(DeclarativeBase):
+    pass
+
+
+#
+#
+# class File(Base):
+#     __tablename__ = "file"
+#     __table_args__ = (PrimaryKeyConstraint("content_hash", name="file_pkey"),)
+#
+#     content_hash: Mapped[str] = mapped_column(String, primary_key=True)
+#     content: Mapped[bytes] = mapped_column(LargeBinary)
+#
+#
+# class InmantaUser(Base):
+#     __tablename__ = "inmanta_user"
+#     __table_args__ = (PrimaryKeyConstraint("id", name="user_pkey"), UniqueConstraint("username", name="user_username_key"))
+#
+#     id: Mapped[uuid.UUID] = mapped_column(UUID, primary_key=True)
+#     username: Mapped[str] = mapped_column(String)
+#     password_hash: Mapped[str] = mapped_column(String)
+#     auth_method: Mapped[str] = mapped_column(Enum("database", "oidc", name="auth_method"))
+#
+#
+class Project(Base):
+    __tablename__ = "project"
+    __table_args__ = (PrimaryKeyConstraint("id", name="project_pkey"), UniqueConstraint("name", name="project_name_key"))
+
+    id: Mapped[uuid.UUID] = mapped_column(UUID, primary_key=True)
+    name: Mapped[str] = mapped_column(String)
+
+    environment: Mapped[List["Environment"]] = relationship("Environment", back_populates="project_")
+
+
+#
+# class Schemamanager(Base):
+#     __tablename__ = "schemamanager"
+#     __table_args__ = (PrimaryKeyConstraint("name", name="schemamanager_pkey"),)
+#
+#     name: Mapped[str] = mapped_column(String, primary_key=True)
+#     installed_versions: Mapped[Optional[list]] = mapped_column(ARRAY(Integer()))
+#
+
+
+class Environment(Base):
```

Comment:
Should we add a comment saying this should only be used for reading the DB and never for writing since it has no validation? (e.g. in the test it gladly accepts the `enable_lsm_expert_mode` setting which is not defined in core)